### PR TITLE
Updated code for mapping pipeline and fastq files download

### PR DESCRIPTION
**Pull request by AnnaMaguza**

### CODE REVIEW

File: fastq_files_download_and_mapping.sh [hunk 1]
```diff
@@ -36,27 +27,28 @@ for accession in "${!samples[@]}"; do
...
+            --soloCBwhitelist $WL \
             --soloCBstart 1 --soloCBlen 16 --soloUMIstart 17 --soloUMIlen 12 --soloStrand Forward \
             --soloUMIdedup 1MM_CR --soloCBmatchWLtype 1MM_multi_Nbase_pseudocounts --soloUMIfiltering MultiGeneUMI_CR \
             --soloCellFilter EmptyDrops_CR --outFilterScoreMin 30 \
             --soloFeatures Gene GeneFull Velocyto --soloOutFileNames output/ features.tsv barcodes.tsv matrix.mtx \
             --soloMultiMappers EM --outReadsUnmapped Fastx
   
   # Delete the initial fastq files
-  rm "$sample_S1_L001_R1_001.fastq" "$sample_S1_L001_R2_001.fastq"
+  rm "${sample}_S1_L001_R1_001.fastq.gz" "${sample}_S1_L001_R2_001.fastq.gz"
```
Comment (cartal) [inline fastq_files_download_and_mapping.sh:51]: To make the code cleaner, you could do `rm *.fastq *.fastq.gz`.

**Review by cartal — CHANGES_REQUESTED** [review]
Hi @AnnaMaguza , I left some changes there. Do you think we can address them? If so, the I can merge.

File: fastq_files_download_and_mapping.sh [hunk 2]
```diff
@@ -36,27 +27,28 @@ for accession in "${!samples[@]}"; do
...
     gunzip "${sample}_${suffix}"
+
   done
   
   # Mapping the fastq files to the genome
   $STAR_CMD --runThreadN 40 \
-            --genomeDir /home/amaguza/data/human_reference_genome/index_file_STAR \
-            --readFilesIn "$sample_name_S1_L001_R1_001.fastq" "$sample_name_S1_L001_R2_001.fastq" \
+            --genomeDir $STAR_INDEX \
+            --readFilesIn "${sample}_S1_L001_R1_001.fastq" "${sample}_S1_L001_R2_001.fastq" \
```
[the remark this answers is not in the document]
Comment (AnnaMaguza) [reply fastq_files_download_and_mapping.sh:41]: I tried both options, but they gave me errors, so I decided to leave decompression as a separate step

File: fastq_files_download_and_mapping.sh [hunk 3]
```diff
@@ -36,27 +27,28 @@ for accession in "${!samples[@]}"; do
...
     
-    # Use axel to download the file with 40 connections
-    axel -n 10 --output="${sample}_${suffix}" "$file_url"
-    
-    # Unzip the files
+    # Use axel to download the file 
+    axel -n $CPU --output="${sample}_${suffix}" "$file_url"
+
+        # Unzip the files
     gunzip "${sample}_${suffix}"
```
[the remark this answers is not in the document]
Comment (AnnaMaguza) [reply fastq_files_download_and_mapping.sh:34]: The decompression on fly gave me errors, so I left it as a separate step

[the remark this answers is not in the document]
Comment (cartal) [reply fastq_files_download_and_mapping.sh:34]: Do you think we can fix it, or should we just move on?

**Pull request merged**